src/connecpy/_server_async.py

```diff
@@ -179,15 +169,9 @@ async def _handle_unary(
                 f"{CONNECT_UNARY_CONTENT_TYPE_PREFIX}{codec.name()}".encode(),
             )
         ]
-        # Compress response if needed
-        if selected_encoding != "identity":
-            compressor = _compression.get_compression(selected_encoding)
-            if compressor:
```

Comment:
I also went ahead and simplified to always set content-encoding, even for `identity`. It simplifies things a lot and can't hurt anything